src/compiler/gcc.rs

```diff
@@ -218,7 +218,7 @@ fn _parse_arguments(arguments: &[OsString],
     let mut outputs = HashMap::new();
     let output = match output_arg {
         // We can't cache compilation that doesn't go to a file
-        None => OsString::from(input.to_str().unwrap().to_owned() + ".o"),
+        None => OsString::from(Path::new(&input).with_extension("o")),
```

Comment:
You just wind up converting this back to a `Path` and `PathBuf` down below, so might as well leave it a `PathBuf` here, and make the `Some` case return a `PathBuf` as well.